tests/biospecimen_diagnosis/test_biospecimen_diagnosis_resources.py

```diff
@@ -0,0 +1,294 @@
+
+        response = self.client.get(url_for(BIO_DIAG_LIST_URL),
+                                   headers=self._api_headers())
+        self.assertEqual(response.status_code, 200)
+        response = json.loads(response.data.decode("utf-8"))
+        content = response.get('results')
+        assert 5 == len(content)
+        assert 5 == BiospecimenDiagnosis.query.count()
+
+    def test_patch(self):
+        """
+        Test updating an existing biospecimen_diagnosis
+        """
+        self._create_all_entities()
+        kf_id = BiospecimenDiagnosis.query.first().kf_id
+
+        # Update existing
+        body = {
+            'external_id': 'bd'
+        }
+        response = self.client.patch(url_for(BIO_DIAG_URL,
+                                             kf_id=kf_id),
+                                     headers=self._api_headers(),
+                                     data=json.dumps(body))
+        # Status code
+        self.assertEqual(response.status_code, 200)
+
+        # Message
+        resp = json.loads(response.data.decode("utf-8"))
+        self.assertIn('biospecimen_diagnosis', resp['_status']['message'])
+        self.assertIn('updated', resp['_status']['message'])
+
+        # Content - check only patched fields are updated
+        res = resp['results']
+        bd = BiospecimenDiagnosis.query.get(kf_id)
+        assert bd.external_id == res['external_id']
+
+    def test_patch_invalid(self):
+        """
+        Test patch a biospecimen_diagnosis by linking a bio and diag from two
+        different participants
+        """
+        self._create_all_entities(link_bio=False)
+
+        # Create a valid one first
+        participants = Participant.query.all()
+        kwargs = {
+            'biospecimen_id': participants[0].biospecimens[0].kf_id,
+            'diagnosis_id': participants[0].diagnoses[0].kf_id
+        }
+        bd = BiospecimenDiagnosis(**kwargs)
+        db.session.add(bd)
+        db.session.commit()
+
+        # Patch it with invalid data
+        patch_kwargs = {
+            'biospecimen_id': participants[1].biospecimens[0].kf_id
+        }
+        # Send get request
+        response = self.client.patch(url_for(BIO_DIAG_URL,
+                                             kf_id=bd.kf_id),
+                                     data=json.dumps(patch_kwargs),
+                                     headers=self._api_headers())
+
+        # Check response status status_code
+        self.assertEqual(response.status_code, 400)
+
+        # Check response content
+        response = json.loads(response.data.decode('utf-8'))
+        response = response['_status']
+        assert 400 == response['code']
+        assert 'cannot be linked with' in response['message']
+        bd = BiospecimenDiagnosis.query.first()
+
+        # Check db - nothing should have updated
+        for k, v in kwargs.items():
+            assert getattr(bd, k) == v
+
+    def test_delete(self):
```

Comment:
Is it worth testing the cascade deletes? Maybe it's already done

Reply:
Yea the test is actually in test_diagnosis_models.py [here](https://github.com/kids-first/kf-api-dataservice/pull/455/files#diff-44639382ba251660ddff1492dbf82c7eL131)